### PR TITLE
[ET-1863] Fix - validate REST `check_in` data before updating

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-1863] 

### 🎥 Artifacts 
🎥 screencast(s): https://www.loom.com/share/50d8cd0c08b6405db5848ca4ef5da3de

### ✔️ Checklist
- [x] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1863]: https://theeventscalendar.atlassian.net/browse/ET-1863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by redscar — DISMISSED** [review]
Looks good. Should we have tests for the new method?

File: src/Tribe/REST/V1/Endpoints/Single_Attendee.php [hunk 1]
```diff
@@ -372,4 +379,36 @@ public function validate_attendee_status( $data, $provider ) {
...
+	 * Validate whether the check_in value is valid for this attendee.
+	 *
+	 * @since TBD Validate check-in data before allowing check-in.
+	 *
+	 * @param $attendee array Attendee data.
+	 * @param $check_in bool Check in value.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public function validate_check_in( array $attendee, bool $check_in ) {
```
[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply src/Tribe/REST/V1/Endpoints/Single_Attendee.php:393]: Yes, the TicketsCommerce statuses are handled. We have been doing this check on the QR endpoint, but it was not in place for the direct attendee data update endpoint. 

https://github.com/the-events-calendar/event-tickets/blob/cd562fea59c5c641424add95acf543d223b483ec/src/Tribe/Status/Manager.php#L360

File: src/Tribe/REST/V1/Endpoints/Single_Attendee.php [hunk 2]
```diff
@@ -372,4 +379,36 @@ public function validate_attendee_status( $data, $provider ) {
...
+		if ( ! tribe_is_truthy( $check_in ) ) {
+			return true;
+		}
+
+		// check if attendee already checked in.
+		if ( tribe_is_truthy( $attendee['check_in'] ) ) {
+			return new WP_Error( 'already-checked-in', __( 'Attendee is already checked in.', 'event-tickets' ), [ 'status' => 400 ] );
```
[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply src/Tribe/REST/V1/Endpoints/Single_Attendee.php:400]: It's for our own endpoint and we don't have prefixes in other places. But I will add them here to start the trend. 👍

**Comment by rafsuntaskin** [comment]
> Looks good. Should we have tests for the new method?

Yes, we should. But I am blocked with `slic` not running `restv1` tests on my local.

**Review by bordoni — APPROVED** [review]

**Pull request merged**